src/main/java/org/jabref/gui/fieldeditors/LinkedFilesEditor.java

```diff
@@ -64,95 +54,14 @@ public LinkedFilesEditor(String fieldName, DialogService dialogService, BibDatab
-                .withOnMouseClickedEvent(this::handleItemMouseClick)
-                .setOnDragDetected(this::handleOnDragDetected)
-                .setOnDragDropped(this::handleOnDragDropped)
-                .setOnDragOver(this::handleOnDragOver);
+                .withOnMouseClickedEvent(this::handleItemMouseClick);
 
         listView.setCellFactory(cellFactory);
 
-        setUpFilesDragAndDrop();
         Bindings.bindContentBidirectional(listView.itemsProperty().get(), viewModel.filesProperty());
         setUpKeyBindings();
     }
 
-    private void setUpFilesDragAndDrop() {
-        listView.setOnDragOver(event -> {
-            if (event.getDragboard().hasFiles()) {
-                event.acceptTransferModes(TransferMode.COPY, TransferMode.LINK);
-            }
-        });
-
-        listView.setOnDragDropped(event -> {
-            Dragboard dragboard = event.getDragboard();
-            boolean success = false;
-            ObservableList<LinkedFileViewModel> items = listView.itemsProperty().get();
-
-            if (dragboard.hasFiles()) {
-                List<LinkedFileViewModel> linkedFiles = dragboard.getFiles().stream().map(File::toPath).map(viewModel::fromFile).collect(Collectors.toList());
-                items.addAll(linkedFiles);
-                success = true;
-            }
-            event.setDropCompleted(success);
-            event.consume();
-        });
-
-    }
-
-    private void handleOnDragOver(LinkedFileViewModel originalItem, DragEvent event) {
-        if ((event.getGestureSource() != originalItem) && event.getDragboard().hasContent(DragAndDropDataFormats.LINKED_FILE)) {
-            event.acceptTransferModes(TransferMode.MOVE);
-        }
-        if (event.getDragboard().hasFiles()) {
-            event.acceptTransferModes(TransferMode.COPY, TransferMode.LINK);
-        }
-    }
-
-    private void handleOnDragDetected(@SuppressWarnings("unused") LinkedFileViewModel linkedFile, MouseEvent event) {
-        LinkedFile selectedItem = listView.getSelectionModel().getSelectedItem().getFile();
-        if (selectedItem != null) {
-            ClipboardContent content = new ClipboardContent();
-            Dragboard dragboard = listView.startDragAndDrop(TransferMode.MOVE);
-            //We have to use the model class here, as the content of the dragboard must be serializable
-            content.put(DragAndDropDataFormats.LINKED_FILE, selectedItem);
-            dragboard.setContent(content);
-        }
-        event.consume();
-    }
-
-    private void handleOnDragDropped(LinkedFileViewModel originalItem, DragEvent event) {
-        Dragboard dragboard = event.getDragboard();
-        boolean success = false;
-
-        ObservableList<LinkedFileViewModel> items = listView.itemsProperty().get();
-
-        if (dragboard.hasContent(DragAndDropDataFormats.LINKED_FILE)) {
-
-            LinkedFile linkedFile = (LinkedFile) dragboard.getContent(DragAndDropDataFormats.LINKED_FILE);
```

Comment:
It is ok to remove the code related to drag & drop of external files, because this is handled by the entry editor as you have noted. However, it should still be possible to resort the linked files using drag & drop.